<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.269">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Robert M Flight">
<meta name="dcterms.date" content="2022-01-15">

<title>Deciphering Life: One Bit at a Time - Recreating Correlation Values from Another Manuscript</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="stylesheet" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../../site_libs/bootstrap/bootstrap-dark.min.css" rel="stylesheet" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<link href="../../site_libs/quarto-contrib/academicons-1.9.2/all.css" rel="stylesheet">
<link href="../../site_libs/quarto-contrib/academicons-1.9.2/size.css" rel="stylesheet">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="../../styles.css">
</head>

<body class="floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">Deciphering Life: One Bit at a Time</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../about.html">
 <span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../random_code_snippets.html">
 <span class="menu-text">RCS</span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/rmflight"><i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="https://orcid.org/0000-0001-8141-7788">
 <span class="menu-text"><i class="ai  ai-orcid" title="" style="color:"></i></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://mastodon.social/@rmflight"><i class="bi bi-mastodon" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
</ul>
              <div class="quarto-toggle-container">
                  <a href="" class="quarto-color-scheme-toggle nav-link" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
              </div>
              <div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default toc-left page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Recreating Correlation Values from Another Manuscript</h1>
            <p class="subtitle lead"></p><p>Documenting my journey trying to recreate some correlations calculated in another manuscript.</p><p></p>
                                <div class="quarto-categories">
                <div class="quarto-category">reproducibility</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>Robert M Flight <a href="https://orcid.org/0000-0001-8141-7788" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">January 15, 2022</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse sidebar-navigation floating overflow-auto">
    <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#background" id="toc-background" class="nav-link active" data-scroll-target="#background">Background</a></li>
  <li><a href="#so-what" id="toc-so-what" class="nav-link" data-scroll-target="#so-what">So What</a></li>
  <li><a href="#my-correlations" id="toc-my-correlations" class="nav-link" data-scroll-target="#my-correlations">My Correlations</a></li>
  <li><a href="#finding-the-data" id="toc-finding-the-data" class="nav-link" data-scroll-target="#finding-the-data">Finding the Data</a></li>
  <li><a href="#new-correlation" id="toc-new-correlation" class="nav-link" data-scroll-target="#new-correlation">New Correlation</a></li>
  <li><a href="#why-no-log-transformation" id="toc-why-no-log-transformation" class="nav-link" data-scroll-target="#why-no-log-transformation">Why No Log-Transformation?</a></li>
  <li><a href="#how-did-i-miss-the-original-data" id="toc-how-did-i-miss-the-original-data" class="nav-link" data-scroll-target="#how-did-i-miss-the-original-data">How Did I Miss the Original Data?</a></li>
  <li><a href="#why-was-it-so-hard-to-reproduce-values" id="toc-why-was-it-so-hard-to-reproduce-values" class="nav-link" data-scroll-target="#why-was-it-so-hard-to-reproduce-values">Why Was it So Hard to Reproduce Values?</a></li>
  <li><a href="#data-files-and-scripts" id="toc-data-files-and-scripts" class="nav-link" data-scroll-target="#data-files-and-scripts">Data Files and Scripts</a></li>
  </ul>
</nav>
</nav>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
    </div>
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content">




<section id="background" class="level2">
<h2 class="anchored" data-anchor-id="background">Background</h2>
<p>I’ve been working on a manuscript for our newish correlation method, information-content-informed Kendall-tau (ICI-Kendalltau, package currently at <span class="citation" data-cites="icikendalltau">(<a href="#ref-icikendalltau" role="doc-biblioref">Flight and Moseley 2021</a>)</span>). As part of that manuscript, we wanted to highlight how well the correlation measure detects outlier samples. Two datasets we are using for that aspect are public, one from The Cancer Genome Atlas and the other from the Barton group. The Barton group and collaborators produced a highly replicated RNAseq yeast dataset, 48 replicates in two conditions, and have used it in various analyses <span class="citation" data-cites="Gierlinski2015 Schurch2016">(<a href="#ref-Gierlinski2015" role="doc-biblioref">Gierliński et al. 2015</a>; <a href="#ref-Schurch2016" role="doc-biblioref">Schurch et al. 2016</a>)</span>.</p>
</section>
<section id="so-what" class="level2">
<h2 class="anchored" data-anchor-id="so-what">So What</h2>
<p>For my manuscript, ideally I want to be able to comment on the outliers found in one of the Barton group manuscripts <span class="citation" data-cites="Gierlinski2015">(<a href="#ref-Gierlinski2015" role="doc-biblioref">Gierliński et al. 2015</a>)</span>. To do that, I need access to one of:</p>
<ul>
<li>the sample-sample correlations themselves,</li>
<li>the counts from each sample</li>
<li>or be able to recreate counts from each sample.</li>
</ul>
<p>Several years ago I was asking about this dataset on twitter when I was using it for another project, because the project ID in the SRA didn’t have a mapping to condition. Dr.&nbsp;Geoff Barton pointed me to a metadata file available on figshare <span class="citation" data-cites="Barton2015">(<a href="#ref-Barton2015" role="doc-biblioref">Barton, Blaxter, Cole, Gharbi, Gierliński, Owen-Hughes, et al. 2015</a>)</span>. This allowed me to generate read counts across biological replicates (which is what I was interested in at the time). However, if I had poked around the figshare project a bit more, I would have likely seen both of the files with read counts for each replicate already available <span class="citation" data-cites="SNF2Barton2015 WTBarton2015">(<a href="#ref-SNF2Barton2015" role="doc-biblioref">Barton, Blaxter, Cole, Gharbi, Gierliński, Schofield, et al. 2015a</a>, <a href="#ref-WTBarton2015" role="doc-biblioref">2015b</a>)</span>. It really sucks that I hadn’t noticed these other files years ago, it would have saved me the effort in remapping and generating gene counts myself, as well as getting really weird correlation values compared to <span class="citation" data-cites="Gierlinski2015">(<a href="#ref-Gierlinski2015" role="doc-biblioref">Gierliński et al. 2015</a>)</span>.</p>
</section>
<section id="my-correlations" class="level2">
<h2 class="anchored" data-anchor-id="my-correlations">My Correlations</h2>
<p>From the demultiplexed read data and the metadata file I found several years ago, I had run RNA-seq mapping software to generate read counts for each sample in each lane, and summed them across lanes. I know how to do these kinds of things, even though I will be the first to admit it is <strong>not</strong> my personal bread and butter analysis (I normally get involved after count generation). With the (few) hints from the manuscript about how <span class="citation" data-cites="Gierlinski2015">(<a href="#ref-Gierlinski2015" role="doc-biblioref">Gierliński et al. 2015</a>)</span> did the correlation calculation (see more below), I don’t get anything close to the range of median correlation values within each class of samples. Which I thought was very, very weird. I implemented a variety of transformation methods and inclusion of missing values for my correlation calculations:</p>
<ul>
<li>logged values (log and log1p)</li>
<li>raw values</li>
<li>removal and inclusion of missing (0) values</li>
</ul>
<p>Even with all of these variations, I could not come close to the same values I found presented in their manuscript. Here, in panel (a) of Figure 2 from <span class="citation" data-cites="Gierlinski2015">(<a href="#ref-Gierlinski2015" role="doc-biblioref">Gierliński et al. 2015</a>)</span>, the median correlations range from 0.7 to 1.</p>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="gierlinski_2015_fig2_cropped.png" class="external img-fluid figure-img" width="950"></p>
<p></p><figcaption class="figure-caption">Figure 2(a) from <span class="citation" data-cites="Gierlinski2015">(<a href="#ref-Gierlinski2015" role="doc-biblioref">Gierliński et al. 2015</a>)</span>. Identifying bad RNA-seq replicates in the WT (left) and Δsnf2 (right) data. The top three panels (a–c) show individual criteria for identifying ‘bad’ replicates which are combined into a quality score (d) in order to identify ‘bad’ replicates in each condition. The identified ‘bad’ replicates are shown as numbered points in each panel. The individual criteria are (a) median correlation coefficient, ri∼⁠, for each replicate i against all other replicates, (b) outlier fraction, fi⁠, calculated as a fraction of genes where the given replicate is more than five-trimmed standard deviations from the trimmed mean and (c) median reduced χ2 of pileup depth, χ∼2i⁠, as a measure of the non-uniformity of read distribution within genes (see also Fig. 3)</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>My ranges, however, were much different.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>out_ranges <span class="ot">=</span> <span class="fu">readRDS</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data_files"</span>, <span class="st">"ranges.rds"</span>))</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(out_ranges<span class="sc">$</span>rmf_ranges, <span class="at">digits =</span> <span class="dv">2</span>, <span class="at">caption =</span> <span class="st">'RMF median correlation ranges. Which is "which" method was used to calculate the correlations.'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="table table-sm table-striped">
<caption>RMF median correlation ranges. Which is “which” method was used to calculate the correlations.</caption>
<thead>
<tr class="header">
<th style="text-align: left;">which</th>
<th style="text-align: left;">sample_class</th>
<th style="text-align: right;">high</th>
<th style="text-align: right;">low</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">log</td>
<td style="text-align: left;">SNF2</td>
<td style="text-align: right;">0.99</td>
<td style="text-align: right;">0.96</td>
</tr>
<tr class="even">
<td style="text-align: left;">log</td>
<td style="text-align: left;">WT</td>
<td style="text-align: right;">0.97</td>
<td style="text-align: right;">0.89</td>
</tr>
<tr class="odd">
<td style="text-align: left;">log_no0</td>
<td style="text-align: left;">SNF2</td>
<td style="text-align: right;">0.99</td>
<td style="text-align: right;">0.95</td>
</tr>
<tr class="even">
<td style="text-align: left;">log_no0</td>
<td style="text-align: left;">WT</td>
<td style="text-align: right;">0.96</td>
<td style="text-align: right;">0.87</td>
</tr>
<tr class="odd">
<td style="text-align: left;">raw</td>
<td style="text-align: left;">SNF2</td>
<td style="text-align: right;">0.98</td>
<td style="text-align: right;">0.96</td>
</tr>
<tr class="even">
<td style="text-align: left;">raw</td>
<td style="text-align: left;">WT</td>
<td style="text-align: right;">0.97</td>
<td style="text-align: right;">0.67</td>
</tr>
<tr class="odd">
<td style="text-align: left;">raw_no0</td>
<td style="text-align: left;">SNF2</td>
<td style="text-align: right;">0.98</td>
<td style="text-align: right;">0.96</td>
</tr>
<tr class="even">
<td style="text-align: left;">raw_no0</td>
<td style="text-align: left;">WT</td>
<td style="text-align: right;">0.97</td>
<td style="text-align: right;">0.67</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>We can see in this table, that my median correlation ranges are not even close to what we can see in the figure.</p>
<p>And the lowest values and samples didn’t seem to be right either.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>out_ranges<span class="sc">$</span>rmf_medians <span class="sc">%&gt;%</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(which <span class="sc">%in%</span> <span class="st">"raw_no0"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(sample_class) <span class="sc">%&gt;%</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(med_cor) <span class="sc">%&gt;%</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">6</span>) <span class="sc">%&gt;%</span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>  knitr<span class="sc">::</span><span class="fu">kable</span>(., <span class="at">digits =</span> <span class="dv">2</span>, <span class="at">caption =</span> <span class="st">"Lowest median correlation values using my own counts."</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="table table-sm table-striped">
<caption>Lowest median correlation values using my own counts.</caption>
<thead>
<tr class="header">
<th style="text-align: left;">sample_id</th>
<th style="text-align: right;">med_cor</th>
<th style="text-align: left;">sample_class</th>
<th style="text-align: left;">which</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">SNF2.7</td>
<td style="text-align: right;">0.96</td>
<td style="text-align: left;">SNF2</td>
<td style="text-align: left;">raw_no0</td>
</tr>
<tr class="even">
<td style="text-align: left;">SNF2.33</td>
<td style="text-align: right;">0.96</td>
<td style="text-align: left;">SNF2</td>
<td style="text-align: left;">raw_no0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">SNF2.27</td>
<td style="text-align: right;">0.96</td>
<td style="text-align: left;">SNF2</td>
<td style="text-align: left;">raw_no0</td>
</tr>
<tr class="even">
<td style="text-align: left;">SNF2.8</td>
<td style="text-align: right;">0.96</td>
<td style="text-align: left;">SNF2</td>
<td style="text-align: left;">raw_no0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">SNF2.48</td>
<td style="text-align: right;">0.96</td>
<td style="text-align: left;">SNF2</td>
<td style="text-align: left;">raw_no0</td>
</tr>
<tr class="even">
<td style="text-align: left;">SNF2.23</td>
<td style="text-align: right;">0.97</td>
<td style="text-align: left;">SNF2</td>
<td style="text-align: left;">raw_no0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">WT.32</td>
<td style="text-align: right;">0.67</td>
<td style="text-align: left;">WT</td>
<td style="text-align: left;">raw_no0</td>
</tr>
<tr class="even">
<td style="text-align: left;">WT.14</td>
<td style="text-align: right;">0.75</td>
<td style="text-align: left;">WT</td>
<td style="text-align: left;">raw_no0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">WT.11</td>
<td style="text-align: right;">0.80</td>
<td style="text-align: left;">WT</td>
<td style="text-align: left;">raw_no0</td>
</tr>
<tr class="even">
<td style="text-align: left;">WT.13</td>
<td style="text-align: right;">0.83</td>
<td style="text-align: left;">WT</td>
<td style="text-align: left;">raw_no0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">WT.16</td>
<td style="text-align: right;">0.87</td>
<td style="text-align: left;">WT</td>
<td style="text-align: left;">raw_no0</td>
</tr>
<tr class="even">
<td style="text-align: left;">WT.12</td>
<td style="text-align: right;">0.91</td>
<td style="text-align: left;">WT</td>
<td style="text-align: left;">raw_no0</td>
</tr>
</tbody>
</table>
</div>
</div>
</section>
<section id="finding-the-data" class="level2">
<h2 class="anchored" data-anchor-id="finding-the-data">Finding the Data</h2>
<p>Finally, during the week of 2021-11-25, I happened across another manuscript on this dataset from 2016 <span class="citation" data-cites="Schurch2016">(<a href="#ref-Schurch2016" role="doc-biblioref">Schurch et al. 2016</a>)</span>, that mentions a GitHub repo that lo and behold had copies of the preprocessed data to the level of gene counts per biological replicate <span class="citation" data-cites="bartongithub2015">(<a href="#ref-bartongithub2015" role="doc-biblioref">Cole et al. 2015</a>)</span>. Awesome!</p>
</section>
<section id="new-correlation" class="level2">
<h2 class="anchored" data-anchor-id="new-correlation">New Correlation</h2>
<p>Using that preprocessed data, I was finally able to get what amounted to identical values of correlation, based on comparing the lowest correlation values with the figure. Great.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(out_ranges<span class="sc">$</span>barton, <span class="at">digits =</span> <span class="dv">2</span>, <span class="at">caption =</span> <span class="st">'Barton median correlation ranges. Which is "which" method was used to calculate the correlations.'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="table table-sm table-striped">
<caption>Barton median correlation ranges. Which is “which” method was used to calculate the correlations.</caption>
<thead>
<tr class="header">
<th style="text-align: left;">which</th>
<th style="text-align: left;">sample_class</th>
<th style="text-align: right;">high</th>
<th style="text-align: right;">low</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">log</td>
<td style="text-align: left;">Snf2</td>
<td style="text-align: right;">0.99</td>
<td style="text-align: right;">0.95</td>
</tr>
<tr class="even">
<td style="text-align: left;">log</td>
<td style="text-align: left;">WT</td>
<td style="text-align: right;">0.99</td>
<td style="text-align: right;">0.97</td>
</tr>
<tr class="odd">
<td style="text-align: left;">log_no0</td>
<td style="text-align: left;">Snf2</td>
<td style="text-align: right;">0.99</td>
<td style="text-align: right;">0.91</td>
</tr>
<tr class="even">
<td style="text-align: left;">log_no0</td>
<td style="text-align: left;">WT</td>
<td style="text-align: right;">0.99</td>
<td style="text-align: right;">0.94</td>
</tr>
<tr class="odd">
<td style="text-align: left;">raw</td>
<td style="text-align: left;">Snf2</td>
<td style="text-align: right;">1.00</td>
<td style="text-align: right;">0.71</td>
</tr>
<tr class="even">
<td style="text-align: left;">raw</td>
<td style="text-align: left;">WT</td>
<td style="text-align: right;">0.99</td>
<td style="text-align: right;">0.77</td>
</tr>
<tr class="odd">
<td style="text-align: left;">raw_no0</td>
<td style="text-align: left;">Snf2</td>
<td style="text-align: right;">1.00</td>
<td style="text-align: right;">0.71</td>
</tr>
<tr class="even">
<td style="text-align: left;">raw_no0</td>
<td style="text-align: left;">WT</td>
<td style="text-align: right;">0.99</td>
<td style="text-align: right;">0.77</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>In this table, for the <strong>raw</strong> correlations, we finally see median correlation ranges that match what is observed in the figure, especially at the low end. Importantly, the Snf2 lowest value is lower than the WT lowest value. So I’m pretty sure I’m getting the correct sample-sample correlations now.</p>
<p>As well, if I look at the lowest sample - sample correlations in each class, the sample IDs match what is in the figure as well, and so do the median sample-sample correlations for those samples!</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>out_ranges<span class="sc">$</span>barton_medians <span class="sc">%&gt;%</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(which <span class="sc">%in%</span> <span class="st">"raw_no0"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(sample_class) <span class="sc">%&gt;%</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(med_cor) <span class="sc">%&gt;%</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">6</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>  knitr<span class="sc">::</span><span class="fu">kable</span>(., <span class="at">digits =</span> <span class="dv">2</span>, <span class="at">caption =</span> <span class="st">"Lowest median correlation values using Barton counts."</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="table table-sm table-striped">
<caption>Lowest median correlation values using Barton counts.</caption>
<thead>
<tr class="header">
<th style="text-align: left;">sample_id</th>
<th style="text-align: right;">med_cor</th>
<th style="text-align: left;">sample_class</th>
<th style="text-align: left;">which</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Snf2.06</td>
<td style="text-align: right;">0.71</td>
<td style="text-align: left;">Snf2</td>
<td style="text-align: left;">raw_no0</td>
</tr>
<tr class="even">
<td style="text-align: left;">Snf2.13</td>
<td style="text-align: right;">0.83</td>
<td style="text-align: left;">Snf2</td>
<td style="text-align: left;">raw_no0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Snf2.35</td>
<td style="text-align: right;">0.96</td>
<td style="text-align: left;">Snf2</td>
<td style="text-align: left;">raw_no0</td>
</tr>
<tr class="even">
<td style="text-align: left;">Snf2.10</td>
<td style="text-align: right;">0.99</td>
<td style="text-align: left;">Snf2</td>
<td style="text-align: left;">raw_no0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Snf2.25</td>
<td style="text-align: right;">0.99</td>
<td style="text-align: left;">Snf2</td>
<td style="text-align: left;">raw_no0</td>
</tr>
<tr class="even">
<td style="text-align: left;">Snf2.24</td>
<td style="text-align: right;">0.99</td>
<td style="text-align: left;">Snf2</td>
<td style="text-align: left;">raw_no0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">WT.21</td>
<td style="text-align: right;">0.77</td>
<td style="text-align: left;">WT</td>
<td style="text-align: left;">raw_no0</td>
</tr>
<tr class="even">
<td style="text-align: left;">WT.25</td>
<td style="text-align: right;">0.86</td>
<td style="text-align: left;">WT</td>
<td style="text-align: left;">raw_no0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">WT.28</td>
<td style="text-align: right;">0.89</td>
<td style="text-align: left;">WT</td>
<td style="text-align: left;">raw_no0</td>
</tr>
<tr class="even">
<td style="text-align: left;">WT.22</td>
<td style="text-align: right;">0.94</td>
<td style="text-align: left;">WT</td>
<td style="text-align: left;">raw_no0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">WT.17</td>
<td style="text-align: right;">0.98</td>
<td style="text-align: left;">WT</td>
<td style="text-align: left;">raw_no0</td>
</tr>
<tr class="even">
<td style="text-align: left;">WT.36</td>
<td style="text-align: right;">0.98</td>
<td style="text-align: left;">WT</td>
<td style="text-align: left;">raw_no0</td>
</tr>
</tbody>
</table>
</div>
</div>
</section>
<section id="why-no-log-transformation" class="level2">
<h2 class="anchored" data-anchor-id="why-no-log-transformation">Why No Log-Transformation?</h2>
<p>One interesting thing about having the correct correlations is discovering that Gierlinski et al didn’t use log-transformed data in their Pearson correlation calculations. This seems unusual to me. All my career in -omics, the one thing I’ve had drilled into me is that doing a linear correlation on data that has proportional error component or variance is a <strong>very bad</strong> idea. Proportional error or variance means the variance increases with increasing mean values, which is definitely true of these count data.</p>
<p>If I had to guess <strong>why</strong> log-transformed values weren’t used, I think it is because of the analysis in the replicate paper about how well the un-transformed values fit a normal distribution vs a log-normal distribution. That figure and caption are provided here for reference.</p>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="gierlinski_2015_fig5.jpeg" class="external img-fluid figure-img" width="950"></p>
<p></p><figcaption class="figure-caption">Figure 5 from <span class="citation" data-cites="Gierlinski2015">(<a href="#ref-Gierlinski2015" role="doc-biblioref">Gierliński et al. 2015</a>)</span>. Goodness-of-fit test results for normal (top panels), log-normal (middle panels) and negative binomial (bottom panels) distributions. Each panel shows the test P-value versus the mean count across replicates. Each dot represents equal-count normalized data from one gene. Panels on the left (a, b, e, f, i, j) show clean data with bad replicates rejected (42 and 44 replicates remaining in WT and Δsnf2, respectively). Panels on the right (c, d, g, h, k, l) show all available data (48 replicates in each condition). Due to the number of bootstraps performed, P-values for the negative-binomial test are limited to ∼10−2. Due to numerical precision of the software library used, P-values from the normal and log-normal tests are limited to ∼10−16. Below these limits data points are marked in orange (light gray in black and white) at the bottom of each panel. Horizontal lines show the Benjamini–Hochberg limit corresponding to the significance of 0.05 for the given dataset. The numbers in the right bottom corner of each panel indicate the number of genes with P-values below the significance limit and the total number of genes</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>Just so you can see the difference using raw and log-space makes (besides the values in the above table), here are two of the replicate samples plotted against each other in raw and log-transformed values.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="fu">theme_set</span>(cowplot<span class="sc">::</span><span class="fu">theme_cowplot</span>())</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>raw_plot <span class="ot">=</span> <span class="fu">readRDS</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data_files"</span>, <span class="st">"raw_plot.rds"</span>))</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>raw_p2 <span class="ot">=</span> raw_plot <span class="sc">+</span> </span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_equal</span>() <span class="sc">+</span></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_abline</span>(<span class="at">slope =</span> <span class="dv">1</span>, <span class="at">color =</span> <span class="st">"red"</span>)</span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>log_plot <span class="ot">=</span> <span class="fu">readRDS</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data_files"</span>, <span class="st">"log_plot.rds"</span>))</span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>log_p2 <span class="ot">=</span> log_plot <span class="sc">+</span></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_equal</span>() <span class="sc">+</span></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_abline</span>(<span class="at">slope =</span> <span class="dv">1</span>, <span class="at">color =</span> <span class="st">"red"</span>)</span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>raw_p2 <span class="sc">+</span> log_p2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="index_files/figure-html/raw_log-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>We can summarize this behavior across <strong>all</strong> the replicate samples, looking at standard deviation (SD) and relative standard deviation (RSD, SD / mean). I took the counts from Barton group after removing outliers based on median correlations, and then calculate the mean, SD, and RSD across all replicates in each of Snf2 and WT.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>data_summary <span class="ot">=</span> <span class="fu">readRDS</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data_files"</span>, <span class="st">"summary.rds"</span>))</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>data_summary <span class="sc">%&gt;%</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>(type <span class="sc">%in%</span> <span class="st">"diff"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> mean, <span class="at">y =</span> var)) <span class="sc">+</span> </span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span> type, <span class="at">ncol =</span> <span class="dv">1</span>, <span class="at">scales =</span> <span class="st">"free_y"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 494 rows containing missing values (`geom_point()`).</code></pre>
</div>
<div class="cell-output-display">
<p><img src="index_files/figure-html/sd_rsd-1.png" class="img-fluid" width="480"></p>
</div>
</div>
<p>As we can see in the above figure, SD increases with increasing mean. Which is basically what we observe in the pairwise plot above. I’m pretty sure I’ve been taught <strong>not</strong> to do Pearson correlation on data with this structure. Interestingly, the RSD does become rather constant after a certain value in mean expression.</p>
</section>
<section id="how-did-i-miss-the-original-data" class="level2">
<h2 class="anchored" data-anchor-id="how-did-i-miss-the-original-data">How Did I Miss the Original Data?</h2>
<p>Back to the question of finding the original data, it turns out I just didn’t look hard enough at the first place that Geoff Barton sent me to. When I started poking around the figshare repo’s from Geoff and others in the group and following links, it was easy to find a copy of the preprocessed data from each condition.</p>
</section>
<section id="why-was-it-so-hard-to-reproduce-values" class="level2">
<h2 class="anchored" data-anchor-id="why-was-it-so-hard-to-reproduce-values">Why Was it So Hard to Reproduce Values?</h2>
<p>Another interesting thing about this endeavor was just how hard it was to reproduce the correlations. <span class="citation" data-cites="Gierlinski2015">(<a href="#ref-Gierlinski2015" role="doc-biblioref">Gierliński et al. 2015</a>)</span> was actually light on details of how the data was processed, and how the correlation was done. For example, the manuscript just says “Pearson correlation”, with basically no other details. Were raw counts used or log-transformed? Were missing (count of 0) values in either sample removed prior to correlation? It was only when I finally happened across the GitHub repo that I finally got the answers I really needed. And for some reason, that manuscript doesn’t mention the GitHub repo, or the data available on figshare. This points to the importance of citing and linking all the resources for a manuscript (or even a blog post!). And I’m not trying to knock on Marek, or Geoff, or Nick on this. In my experience, they post open things, and provide lots of data. But this highlights just how hard it becomes to recreate something if even a little piece of the data is missing.</p>
</section>
<section id="data-files-and-scripts" class="level2">
<h2 class="anchored" data-anchor-id="data-files-and-scripts">Data Files and Scripts</h2>
<p>The rds files and a processing script to calculate the correlations and generate the plots are all available on the blog directory on GitHub <span class="citation" data-cites="flight_post">(<a href="#ref-flight_post" role="doc-biblioref">Flight 2022</a>)</span>.</p>



</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" role="doc-bibliography">
<div id="ref-Barton2015" class="csl-entry" role="doc-biblioentry">
Barton, Geoffrey, Mark Blaxter, Christian Cole, Karim Gharbi, Marek Gierliński, Tom Owen-Hughes, Pieta Schofield, et al. 2015. <span>“<span class="nocase">Metadata for a highly replicated two-condition yeast RNAseq experiment.</span>”</span> July. <a href="https://doi.org/10.6084/m9.figshare.1416210.v3">https://doi.org/10.6084/m9.figshare.1416210.v3</a>.
</div>
<div id="ref-SNF2Barton2015" class="csl-entry" role="doc-biblioentry">
Barton, Geoffrey, Mark Blaxter, Christian Cole, Karim Gharbi, Marek Gierliński, Pieta Schofield, Nick Schurch, Vijender Singh, and Nicola Wrobel. 2015a. <span>“<span class="nocase">SNF2 knock-out yeast gene read counts from a 48 replicate experiment</span>,”</span> May. <a href="https://doi.org/10.6084/m9.figshare.1425502.v1">https://doi.org/10.6084/m9.figshare.1425502.v1</a>.
</div>
<div id="ref-WTBarton2015" class="csl-entry" role="doc-biblioentry">
———. 2015b. <span>“<span class="nocase">Wild-type yeast gene read counts from 48 replicate experiment</span>,”</span> May. <a href="https://doi.org/10.6084/m9.figshare.1425503.v1">https://doi.org/10.6084/m9.figshare.1425503.v1</a>.
</div>
<div id="ref-bartongithub2015" class="csl-entry" role="doc-biblioentry">
Cole, Chris, Nick Schurch, Rayna M Harris, and Marek Gierliński. 2015. <span>“profDGE48.”</span> <a href="https://github.com/bartongroup/profDGE48">https://github.com/bartongroup/profDGE48</a>.
</div>
<div id="ref-flight_post" class="csl-entry" role="doc-biblioentry">
Flight, Robert M. 2022. <span>“Post Files.”</span> <a href="https://github.com/rmflight/researchBlog_distill/tree/main/_posts/2022-01-15-recreating-correlation-values-from-another-manuscript">https://github.com/rmflight/researchBlog_distill/tree/main/_posts/2022-01-15-recreating-correlation-values-from-another-manuscript</a>.
</div>
<div id="ref-icikendalltau" class="csl-entry" role="doc-biblioentry">
Flight, Robert M, and Hunter NB Moseley. 2021. <span>“ICIKendallTau.”</span> <a href="https://github.com/MoseleyBioinformaticsLab/ICIKendallTau">https://github.com/MoseleyBioinformaticsLab/ICIKendallTau</a>.
</div>
<div id="ref-Gierlinski2015" class="csl-entry" role="doc-biblioentry">
Gierliński, Marek, Christian Cole, Pietà Schofield, Nicholas J. Schurch, Alexander Sherstnev, Vijender Singh, Nicola Wrobel, et al. 2015. <span>“<span class="nocase">Statistical models for RNA-seq data derived from a two-condition 48-replicate experiment</span>.”</span> <em>Bioinformatics</em> 31 (22): 3625–30. <a href="https://doi.org/10.1093/bioinformatics/btv425">https://doi.org/10.1093/bioinformatics/btv425</a>.
</div>
<div id="ref-Schurch2016" class="csl-entry" role="doc-biblioentry">
Schurch, Nicholas J., Pietá Schofield, Marek Gierliński, Christian Cole, Alexander Sherstnev, Vijender Singh, Nicola Wrobel, et al. 2016. <span>“How Many Biological Replicates Are Needed in an RNA-Seq Experiment and Which Differential Expression Tool Should You Use?”</span> <em>RNA</em> 22 (6): 839–51. <a href="https://doi.org/10.1261/rna.053959.115">https://doi.org/10.1261/rna.053959.115</a>.
</div>
</div></section><section class="quarto-appendix-contents"><h2 class="anchored quarto-appendix-heading">Reuse</h2><div quarto-reuse="quarto-reuse" class="quarto-appendix-contents"><a rel="license" href="https://creativecommons.org/licenses/by/4.0/">https://creativecommons.org/licenses/by/4.0/</a></div></section><section class="quarto-appendix-contents"><h2 class="anchored quarto-appendix-heading">Citation</h2><div><div class="quarto-appendix-secondary-label">BibTeX citation:</div><pre class="sourceCode code-with-copy quarto-appendix-bibtex"><code class="sourceCode bibtex">@online{mflight2022,
  author = {Robert M Flight},
  title = {Recreating {Correlation} {Values} from {Another}
    {Manuscript}},
  date = {2022-01-15},
  url = {https://rmflight.github.io/posts/2022-01-15-recreating-correlation-values-from-another-manuscript},
  langid = {en}
}
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre><div class="quarto-appendix-secondary-label">For attribution, please cite this work as:</div><div id="ref-mflight2022" class="csl-entry quarto-appendix-citeas" role="doc-biblioentry">
Robert M Flight. 2022. <span>“Recreating Correlation Values from Another
Manuscript.”</span> January 15, 2022. <a href="https://rmflight.github.io/posts/2022-01-15-recreating-correlation-values-from-another-manuscript">https://rmflight.github.io/posts/2022-01-15-recreating-correlation-values-from-another-manuscript</a>.
</div></div></section></div></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  let localAlternateSentinel = 'alternate';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<script src="https://giscus.app/client.js" data-repo="rmflight/researchblog" data-repo-id="R_kgDOIjZROQ" data-category="Announcements" data-category-id="DIC_kwDOIjZROc4CS6Xp" data-mapping="pathname" data-reactions-enabled="1" data-emit-metadata="0" data-input-position="top" data-theme="dark" data-lang="en" crossorigin="anonymous" async="">
</script>
</div> <!-- /content -->



</body></html>